pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -103,12 +110,7 @@ func (am *externalAlertmanager) GetSilence(ctx context.Context, silenceID string
 		return apimodels.GettableSilence{}, err
 	}
 
-	if res != nil {
```

Comment:
We would need to add this guard to almost every method, I think it's overkill. I'd say it's safe to assume that the `go-swagger` package won't introduce a breaking change like returning a nil value when no error occurred.